internal_workflow.go

```diff
@@ -327,6 +333,7 @@ func (d *syncWorkflowDefinition) Execute(env workflowEnvironment, input []byte)
 	d.rootCtx = WithValue(background, workflowEnvironmentContextKey, env)
 	var resultPtr *workflowResult
 	d.rootCtx = WithValue(d.rootCtx, workflowResultContextKey, &resultPtr)
+	d.rootCtx = WithValue(d.rootCtx, changeVersionsContextKey, make(map[string]*changeVersion))
```

Comment:
why this needs to be on rootCtx? can the versions map be part of workflowEnvironment, just like the sideEffectResult is part of workflowEnvironmentImpl.